server/src/main/java/org/elasticsearch/index/search/stats/ShardSearchAPMMetrics.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.index.search.stats;
+
+import org.elasticsearch.index.shard.SearchOperationListener;
+import org.elasticsearch.search.internal.SearchContext;
+import org.elasticsearch.telemetry.metric.LongHistogram;
+import org.elasticsearch.telemetry.metric.MeterRegistry;
+import org.elasticsearch.threadpool.ThreadPool;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public final class ShardSearchAPMMetrics implements SearchOperationListener {
+
+    public static final String SEARCH_PHASES_DURATION_METRIC = "es.search.shards.phases.duration.histogram";
```

Comment:
nit: for consistency with the other metrics should we use latency instead of duration?

Reply:
I'm ok with that - though latency has a networking meaning to me, that's why I changed that name.

Reply:
ok duration is fine, it's a different metric now